### PR TITLE
Branch delete index

### LINKED ISSUE #122: [PE-D] Delete entry with the same name is not optimized
Step to reproduce:
Have two entries with the same name called consultation
edelete consultation
Reason:
Although only the first occurrence will be deleted, when user want to delete the second occurrence or later ones, he will have to delete all the entries with the same name ahead

-------------
Labels: `severity.High` `type.FeatureFlaw`
original: ljhgab/ped#8

### LINKED ISSUE #125: [PE-D] edelete argument error not match
Step to reproduce:
edelete 1
Expected:
Should inform the user that name should be used to edelete an entry
Invalid arguments! edelete NAME
Actual:
There is no such entry with that name!

-------------
Labels: `severity.Medium` `type.FunctionalityBug`
original: ljhgab/ped#11

### LINKED ISSUE #137: [PE-D] Inconvenient edelete command
#### Issue description
- The current implementation of edelete only deletes the first matching entry from the list, which could be inconvenient. If there are two entries with the same name, there is no way to delete the second entry without deleting the first entry first. 

#### Steps to reproduce the issue

1. eadd n/meeting sd/2021-02-15 21:00 ed/2021-02-15 23:00
2. eadd n/meeting sd/2022-02-15 21:00 ed/2022-02-15 23:00
3. edelete meeting (intending to delete the second entry)

#### What's the expected result?

- Second entry being deleted while the first entry remains unchanged in the system.

#### What's the actual result?

- First entry needs to be deleted first otherwise the second entry could never be deleted.

#### Additional details / screenshot

- 

-------------
Labels: `severity.High` `type.FunctionalityBug`
original: NiniJiaying/ped#6

### LINKED ISSUE #144: [PE-D] Bug: deletion
Entries are allowed to have the same exact names, but deleting deletes by name as well. According to the UG it deletess the earliest one, but what if I want to delete the 3rd entry of the same name. I would have to delete all the 1-3 entries instead

-------------
Labels: `severity.High` `type.FunctionalityBug`
original: VisnuRavi/ped#6

### LINKED ISSUE #157: [PE-D] Deleting entries with same name
When the name of the entry is identical, cannot specify which entry to delete since edelete only takes in name as pararmeter

-------------
Labels: `severity.Medium` `type.FeatureFlaw`
original: maikongeh/ped#1

### PR DESCRIPTION
**Pull request by lrj689**
Parameter for edelete changed from name to index, added test for edelete aswell as other testing utilities

### CODE REVIEW

File: src/main/java/seedu/address/commons/core/Messages.java [hunk 1]
```diff
@@ -6,7 +6,7 @@
 public class Messages {
     public static final String MESSAGE_UNKNOWN_COMMAND = "Unknown command";
     public static final String MESSAGE_INVALID_COMMAND_FORMAT = "Invalid command format! \n%1$s";
-    public static final String MESSAGE_NO_SUCH_ENTRY = "There is no such entry with that name!";
+    public static final String MESSAGE_NO_SUCH_ENTRY = "The entry index provided is invalid";
```
Comment (nicholastanvis) [inline src/main/java/seedu/address/commons/core/Messages.java]: might wanna change this to MESSAGE_INVALID_ENTRY_INDEX

File: src/main/java/seedu/address/logic/commands/DeleteEntryCommand.java [hunk 2]
```diff
@@ -5,37 +5,47 @@
...
 
 import seedu.address.commons.core.Messages;
+import seedu.address.commons.core.index.Index;
 import seedu.address.logic.commands.exceptions.CommandException;
 import seedu.address.model.Model;
 import seedu.address.model.entry.Entry;
-import seedu.address.model.entry.EntryNameContainsKeywordsPredicate;
 
+/**
+ * Deletes an entry identified using it's displayed index from teaching assistant.
```
Comment (nicholastanvis) [inline src/main/java/seedu/address/logic/commands/DeleteEntryCommand.java]: should be "its"
xd

if can, capitalise Teaching Assistant

File: src/test/java/seedu/address/logic/commands/DeleteCommandTest.java [hunk 3]
```diff
@@ -5,8 +5,8 @@
 import static seedu.address.logic.commands.CommandTestUtil.assertCommandFailure;
 import static seedu.address.logic.commands.CommandTestUtil.assertCommandSuccess;
 import static seedu.address.logic.commands.CommandTestUtil.showPersonAtIndex;
-import static seedu.address.testutil.TypicalIndexes.INDEX_FIRST_PERSON;
-import static seedu.address.testutil.TypicalIndexes.INDEX_SECOND_PERSON;
+import static seedu.address.testutil.TypicalIndexes.INDEX_FIRST;
```
Comment (nicholastanvis) [inline src/test/java/seedu/address/logic/commands/DeleteCommandTest.java]: the plural of index is indices

File: src/test/java/seedu/address/testutil/EntryBuilder.java [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+import seedu.address.model.util.SampleDataUtil;
+
+/**
+ * A utility class to help with building entry objects.
+ */
+public class EntryBuilder {
+
+    public static final String DEFAULT_ENTRY_NAME = "Consultation";
+    public static final String DEFAULT_START_DATE = "2021-04-05 17:00";
+    public static final String DEFAULT_END_DATE = "2021-04-05 19:00";
```
Comment (nicholastanvis) [inline src/test/java/seedu/address/testutil/EntryBuilder.java:18]: should we have DEFAULT_TAGS so we can feed it into the ctor in line 30?

File: src/test/java/seedu/address/testutil/TypicalEntries.java [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+    public static final Entry EXTRA_CLASS = new EntryBuilder().withEntryName("Extra class")
+            .withStartDate("2021-02-02 17:00").withEndDate("2021-02-02 18:30").withTags("Math").build();
+
+    public static final Entry SHORT_QUIZ = new EntryBuilder().withEntryName("Short Geo Quiz")
+            .withStartDate("2021-02-02 14:30").withEndDate("2021-02-02 15:15").withTags("Geography", "21A01").build();
+
+    public static final Entry DO_STUFF = new EntryBuilder().withEntryName("Go do something important")
+            .withStartDate("2021-02-04 09:00").withEndDate("2021-02-04 09:30").withTags("Important", "Stuff").build();
+
+    private TypicalEntries() {}
```
Comment (nicholastanvis) [inline src/test/java/seedu/address/testutil/TypicalEntries.java]: what is this line for?

File: src/test/java/seedu/address/testutil/TypicalIndexes.java [hunk 6]
```diff
@@ -6,7 +6,7 @@
  * A utility class containing a list of {@code Index} objects to be used in tests.
  */
 public class TypicalIndexes {
-    public static final Index INDEX_FIRST_PERSON = Index.fromOneBased(1);
-    public static final Index INDEX_SECOND_PERSON = Index.fromOneBased(2);
-    public static final Index INDEX_THIRD_PERSON = Index.fromOneBased(3);
+    public static final Index INDEX_FIRST = Index.fromOneBased(1);
+    public static final Index INDEX_SECOND = Index.fromOneBased(2);
+    public static final Index INDEX_THIRD = Index.fromOneBased(3);
```
Comment (nicholastanvis) [inline src/test/java/seedu/address/testutil/TypicalIndexes.java]: based and redpilled

**Review by nicholastanvis — CHANGES_REQUESTED** [review]

**Review by goatygoatygoat — APPROVED** [review]

**Review by nicholastanvis — APPROVED** [review]

**Review by tsh22 — APPROVED** [review]

**Pull request merged**